server/src/main/java/org/elasticsearch/cluster/NodeUsageStatsForThreadPoolsCollectorImpl.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.node.usage.NodeUsageStatsForThreadPoolsAction;
+import org.elasticsearch.action.admin.cluster.node.usage.TransportNodeUsageStatsForThreadPoolsAction;
+import org.elasticsearch.client.internal.Client;
+
+import java.util.Map;
+
+// TODO: replace the NodeUsageStatsForThreadPoolsCollector interface with this class.
+public class NodeUsageStatsForThreadPoolsCollectorImpl implements NodeUsageStatsForThreadPoolsCollector {
+
+    @Override
+    public void collectUsageStats(Client client, ActionListener<Map<String, NodeUsageStatsForThreadPools>> listener) {
+        client.execute(
+            TransportNodeUsageStatsForThreadPoolsAction.TYPE,
+            new NodeUsageStatsForThreadPoolsAction.Request(),
+            listener.map(response -> response.getAllNodeUsageStatsForThreadPools())
```

Comment:
We don't check for failures here (e.g. if one node failed) I wonder if it makes sense to pass a partial result through.

Reply:
I think we could use the last stats sent for any node that fails to respond, and have the *Monitor log a a debug message about the missing response. I'd expect the cluster has other issues if a node fails to respond, or perhaps a race with removing a node. I can't think of any obvious harm in using stats that are a little stale. We could go as far as logging a warning if a node is not updated after X rounds -- keep some kind of timestamp of last stats update per node.